packages/cadl-ranch-specs/http/models/property-types/main.tsp

```diff
@@ -170,16 +170,78 @@ model CollectionsModelProperty is ModelTemplate<InnerModel[]>;
 interface CollectionsModel
   extends ModelOperations<CollectionsModelProperty, "[{'property': 'hello'}, {'property': 'world'}]"> {}
 
+// Test a model with a nullable collection property
+@doc("Model with collection with nullable values")
+model CollectionsNullableItemProperty is ModelTemplate<(int32 | null)[]>;
+@route("/collections/nullable-item")
+@operationGroup
+interface CollectionsNullableItem extends ModelOperations<CollectionsNullableItemProperty, "[1, null, 3]"> {}
+
 // Test a model with a string dictionary property
 @doc("Model with dictionary string properties")
 model DictionaryStringProperty is ModelTemplate<Record<string>>;
 @route("/dictionary/string")
 @operationGroup
 interface DictionaryString extends ModelOperations<DictionaryStringProperty, "{'k1': 'hello', 'k2': 'world'}"> {}
 
+// Test a model with a nullable dictionary property
+@doc("Model with collection with nullable values")
+model DictionaryNullableValueProperty is ModelTemplate<Record<string | null>>;
+@route("/dictionary/nullable-value")
+@operationGroup
+interface DictionaryNullableValue
+  extends ModelOperations<DictionaryNullableValueProperty, "{'k1': 'hello', 'k2': 'world', 'k3': null}"> {}
+
 // Test a model with a never property
 @doc("Model with a property never. (This property should not be included).")
 model NeverProperty is ModelTemplate<never>;
 @route("/never")
 @operationGroup
 interface Never extends ModelOperations<NeverProperty, "<don't include this property>"> {}
+
+// Test a model with a model property which could be nullable
+@doc("Model with nullable properties")
+model ModelWithNullableProperty {
+  @doc("id property")
+  id: string;
+
+  @doc("nullable property")
+  property: string | null;
+}
+
+@doc("Model with a property null. The null value will be set in application/merge-patch+json case.")
+@route("/nullable-property")
+@operationGroup
+interface ModelNullableProperty {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing"
+  @scenario
+  @scenarioDoc("""
+Expected response body::
+  ```json
+  {
+    "id": 1,
+    "property": "foo"
```

Comment:
It is `ModelNullableProperty` test, do you want property="foo" or property=null?

Reply:
I design the case the property could be string or null, so here in `get` method it would be a foo string, and in `patch` method it would be a explicit null value.